.github/workflows/cypress-e2e-reporting-test.yml

```diff
@@ -0,0 +1,180 @@
+          target: plugin-artifacts/
+          filename: reports-scheduler.zip
+
+      - name: Download OpenSearch
+        uses: peternied/download-file@v2
+        with:
+          url: https://artifacts.opensearch.org/snapshots/core/opensearch/${{ env.OPENSEARCH_VERSION }}-SNAPSHOT/opensearch-min-${{ env.OPENSEARCH_VERSION }}-SNAPSHOT-linux-x64-latest.tar.gz
+
+      - name: Extract OpenSearch
+        run: |
+          tar -xzf opensearch-*.tar.gz
+          rm -f opensearch-*.tar.gz
+        shell: bash
+
+      - name: Install Job Scheduler
+        run: |
+          /bin/bash -c "yes | ./opensearch-${{ env.OPENSEARCH_VERSION }}-SNAPSHOT/bin/opensearch-plugin install file:$(pwd)/plugin-artifacts/job-scheduler.zip"
+        shell: bash
+    
+      - name: Install Reports Scheduler
+        run: |
+          /bin/bash -c "yes | ./opensearch-${{ env.OPENSEARCH_VERSION }}-SNAPSHOT/bin/opensearch-plugin install file:$(pwd)/plugin-artifacts/reports-scheduler.zip"
+        shell: bash
+
+      - name: Run OpenSearch
+        run: |
+          /bin/bash -c "./opensearch-${{ env.OPENSEARCH_VERSION }}-SNAPSHOT/bin/opensearch &"
+          sleep 30
+        shell: bash
+
+      - name: Check OpenSearch Running on Linux
+        if: ${{ runner.os != 'Windows'}}
+        run: curl http://localhost:9200/
+        shell: bash
+
+      - name: Show OpenSearch Logs
+        if: always()
+        run: cat ./opensearch-${{ env.OPENSEARCH_VERSION }}-SNAPSHOT/logs/opensearch.log
+        shell: bash
+
+      - name: Checkout OpenSearch Dashboards
+        uses: actions/checkout@v2
+        with:
+          path: OpenSearch-Dashboards
+          repository: opensearch-project/OpenSearch-Dashboards
+          ref: ${{ env.OPENSEARCH_DASHBOARDS_VERSION }}
+          fetch-depth: 0
+          filter: |
+            cypress
+            test
+
+      - name: Checkout Dashboards Reporting Plugin in OpenSearch Dashboards Plugins Dir
+        uses: actions/checkout@v2
+        with:
+            path: OpenSearch-Dashboards/plugins/${{ env.PLUGIN_NAME }}
+
+      - id: tool-versions
+        run: |
+          echo "node_version=$(cat .node-version)" >> $GITHUB_OUTPUT
+          echo "yarn_version=$(jq -r '.engines.yarn' package.json)" >> $GITHUB_OUTPUT
+        working-directory: OpenSearch-Dashboards
+        shell: bash
+
+      - uses: actions/setup-node@v1
+        with:
+          node-version: ${{ steps.tool-versions.outputs.node_version }}
+          registry-url: 'https://registry.npmjs.org'
+
+      - name: Setup Opensearch Dashboards
+        run: |
+          npm uninstall -g yarn
+          echo "Installing yarn ${{ steps.tool-versions.outputs.yarn_version }}"
+          npm i -g yarn@${{ steps.tool-versions.outputs.yarn_version }}
+          yarn cache clean
+        working-directory: OpenSearch-Dashboards
+        shell: bash
+
+      - name: OpenSearch Dashboards Plugin Bootstrap
```

Comment:
Here goes a super wired error: https://github.com/opensearch-project/dashboards-reporting/actions/runs/7228046528/job/19696789121?pr=262#step:17:154
I can reproduce it on my local only during the first run of yarn osd bootstrap . And if I re-run it after it fails, then it works again..

So in the current commit, I added max 3 re-tries for the bootstrap step, and it got the CI works.. but in the 1st try, you can see the exactly the same error: https://github.com/opensearch-project/dashboards-reporting/actions/runs/7228507642/job/19698061142?pr=262#step:17:166

Reply:
Transferring some of the conversation between me and @ps48 here:

We think it is fine to do this CI work here first and then we address the above bootstrap issue, since we have a current workaround on that, so that it is not a hard blocker. I will file an issue to track that.

Reply:
Tracking issue is here: https://github.com/opensearch-project/dashboards-reporting/issues/263